src/mpid/ch4/src/ch4_comm.h

```diff
@@ -146,6 +148,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Comm_create_hook(MPIR_Comm * comm)
     int max_n_avts;
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_COMM_CREATE_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_COMM_CREATE_HOOK);
+
+    MPIDI_CH4_collective_selection_init(comm);
```

Comment:
`MPIDI_collective_selection_init` should be sufficient. the CH4 in function name is mostly redundant. we may scrub all instances from the rest of the device.